File: verification/cpp/section6_riemann_zeros/main.cpp
#include <cmath>
#include <iostream>
bool check(const std::string& n, double e, double a) {
    bool p = std::abs(e-a) < 1e-10;
    std::cout << "  [" << (p?"PASS":"FAIL") << "] " << n << "\n";
    return p;
}
int main() {
    std::cout << "=== Section 6: Riemann Zeros (C++) ===\n";
    double g1 = 14.134725141734693, g2 = 21.022039638771555, g3 = 25.010857580145688;
    bool all = true;
    all &= check("gamma_1 > 0", 1, g1 > 0 ? 1 : 0);
    all &= check("gamma_1 < gamma_2", 1, g1 < g2 ? 1 : 0);
    all &= check("gamma_2 < gamma_3", 1, g2 < g3 ? 1 : 0);
    std::cout << "JSON: {\"section\": 6, \"language\": \"cpp\", \"values\": {\"gamma_1\": " << g1 << "}, \"all_passed\": " << (all?"true":"false") << "}\n";
    return all ? 0 : 1;
}

File: verification/rust/section6_riemann_zeros/src/main.rs
fn check(name: &str, expected: f64, actual: f64) -> bool {
    let pass = (expected - actual).abs() < 1e-10;
    println!("  [{}] {}: expected={:.15e}, actual={:.15e}",
        if pass {"PASS"} else {"FAIL"}, name, expected, actual);
    pass
}
fn main() {
    println!("=== Section 6: Riemann Zeros (Rust) ===");
    let g1 = 14.134725141734693_f64;
    let g2 = 21.022039638771555_f64;
    let g3 = 25.010857580145688_f64;
    let mut all_pass = true;
    all_pass &= check("gamma_1 > 0", 1.0, if g1 > 0.0 {1.0} else {0.0});
    all_pass &= check("gamma_1 < gamma_2", 1.0, if g1 < g2 {1.0} else {0.0});
    all_pass &= check("gamma_2 < gamma_3", 1.0, if g2 < g3 {1.0} else {0.0});
    println!("JSON: {{\"section\": 6, \"language\": \"rust\", \"values\": {{\"gamma_1\": {:.15e}, \"gamma_2\": {:.15e}, \"gamma_3\": {:.15e}}}, \"all_passed\": {}}}",
        g1, g2, g3, all_pass);
    if !all_pass { std::process::exit(1); }
}
